diff --git a/vendor/knative.dev/reconciler-test/pkg/feature/feature.go b/vendor/knative.dev/reconciler-test/pkg/feature/feature.go
index c245012ff..933f50e79 100644
--- a/vendor/knative.dev/reconciler-test/pkg/feature/feature.go
+++ b/vendor/knative.dev/reconciler-test/pkg/feature/feature.go
@@ -229,7 +229,7 @@ func DeleteResources(ctx context.Context, t T, refs []corev1.ObjectReference) er
 
 	var lastResource corev1.ObjectReference // One still present resource
 
-	err := wait.Poll(time.Second, 4*time.Minute, func() (bool, error) {
+	err := wait.Poll(time.Second, 8*time.Minute, func() (bool, error) {
 		for _, ref := range refs {
 			gv, err := schema.ParseGroupVersion(ref.APIVersion)
 			if err != nil {
