diff --git a/vendor/knative.dev/eventing/test/rekt/features/trigger/feature.go b/vendor/knative.dev/eventing/test/rekt/features/trigger/feature.go
index 4896fd1c7..219e48535 100644
--- a/vendor/knative.dev/eventing/test/rekt/features/trigger/feature.go
+++ b/vendor/knative.dev/eventing/test/rekt/features/trigger/feature.go
@@ -69,7 +69,8 @@ func TriggerDependencyAnnotation() *feature.Feature {
 	// Install the trigger
 	f.Setup("install trigger", trigger.Install(triggerName, brokerName, cfg...))
 
-	f.Setup("trigger goes ready", trigger.IsReady(triggerName))
+	// trigger won't go ready until after the pingsource exists, because of the dependency annotation
+	f.Requirement("trigger goes ready", trigger.IsReady(triggerName))
 
 	f.Requirement("install pingsource", func(ctx context.Context, t feature.T) {
 		brokeruri, err := broker.Address(ctx, brokerName)
diff --git a/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.go b/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.go
index dc59d58ba..e6ea82a71 100644
--- a/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.go
+++ b/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.go
@@ -124,18 +124,14 @@ func WithSubscriberFromDestination(dest *duckv1.Destination) manifest.CfgFn {
 // WithAnnotations adds annotations to the trigger
 func WithAnnotations(annotations map[string]interface{}) manifest.CfgFn {
 	return func(cfg map[string]interface{}) {
-		if _, set := cfg["ceOverrides"]; !set {
-			cfg["ceOverrides"] = map[string]interface{}{}
+		if _, set := cfg["annotations"]; !set {
+			cfg["annotations"] = map[string]string{}
 		}
-		ceOverrides := cfg["ceOverrides"].(map[string]interface{})
 
 		if annotations != nil {
-			if _, set := ceOverrides["annotations"]; !set {
-				ceOverrides["annotations"] = map[string]interface{}{}
-			}
-			ceExt := ceOverrides["annotations"].(map[string]interface{})
+			annotation := cfg["annotations"].(map[string]string)
 			for k, v := range annotations {
-				ceExt[k] = v
+				annotation[k] = v.(string)
 			}
 		}
 	}
diff --git a/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.yaml b/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.yaml
index 4521988db..77d02ad19 100644
--- a/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.yaml
+++ b/vendor/knative.dev/eventing/test/rekt/resources/trigger/trigger.yaml
@@ -20,7 +20,7 @@ metadata:
   {{ if .annotations }}
   annotations:
   {{ range $key, $value := .annotations }}
-    {{ $key }}: {{ $value }}
+    {{ $key }}: '{{ $value }}'
   {{ end }}
   {{ end }}
 spec:
